ext/dcos-installer/dcos_installer/cli_dispatcher.py

```diff
@@ -0,0 +1,171 @@
+import asyncio
+import coloredlogs
+import logging
+import sys
+
+from dcos_installer import async_server, action_lib, backend
+from dcos_installer.action_lib.prettyprint import print_header, PrettyPrint
+from dcos_installer.installer_analytics import InstallerAnalytics
+
+from ssh.utils import AbstractSSHLibDelegate
+
+log = logging.getLogger(__name__)
+installer_analytics = InstallerAnalytics()
+
+
+class CliDelegate(AbstractSSHLibDelegate):
+    def on_update(self, future, callback_called):
+        chain_name, result_object, host = future.result()
+        callback_called.set_result(True)
+        log.debug('on_update executed for {}'.format(chain_name))
+
+    def on_done(self, name, result, host_status=None):
+        print_header('STAGE {}'.format(name))
+
+    def prepare_status(self, name, nodes):
+        pass
+
+
+def run_loop(action, options):
+    assert callable(action)
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+
+    print_header('START {}'.format(action.__name__))
+    try:
+        config = backend.get_config()
+        cli_delegate = CliDelegate()
+        result = loop.run_until_complete(action(config, block=True, async_delegate=cli_delegate, options=options))
+        pp = PrettyPrint(result)
+        pp.stage_name = action.__name__
+        pp.beautify('print_data')
+
+    finally:
+        loop.close()
+    exitcode = 0
+    for host_result in result:
+        for command_result in host_result:
+            for host, process_result in command_result.items():
+                if process_result['returncode'] != 0:
+                    exitcode += 1
+    print_header('ACTION {} COMPLETE'.format(action.__name__))
+    pp.print_summary()
+    return exitcode
+
+
+def tall_enough_to_ride():
+    choices_true = ['yes', 'y']
+    choices_false = ['no', 'n']
+    while True:
+        do_uninstall = input('This will uninstall DC/OS on your cluster. You may need to manually remove '
+                             '/var/lib/zookeeper in some cases after this completes, please see our documentation '
+                             'for details. Are you ABSOLUTELY sure you want to proceed? [ (y)es/(n)o ]: ')
+        if do_uninstall.lower() in choices_true:
+            return
+        elif do_uninstall.lower() in choices_false:
+            sys.exit(1)
+        else:
+            log.error('Choices are [y]es or [n]o. "{}" is not a choice'.format(do_uninstall))
+
+
+def log_warn_only():
+    """Drop to warning level and down to get around gen.generate() log.info
+    output"""
+    coloredlogs.install(
+        level='WARNING',
+        level_styles={
```

Comment:
This code could be replaced with a call to https://github.com/dcos/dcos/pull/43/files#diff-b1a486deebfabe07c0790a5737a04c79R12 that sets level = 'WARNING'. Right now most of it is just copy / paste.

Probably makes sense to change setup_logger to take the level as the parameter (rather than an options object). IT doesn't really use anything but that one field. In the one case we call it with an options object, just pass in (`'DEBUG' if options.verbose else 'INFO'`)